utils.go

```diff
@@ -325,11 +325,20 @@ func GenerateRandomStringURLSafe(n int) (string, error) {
 // number generator fails to function correctly, in which
 // case the caller should not continue.
 func GenerateRandomStringDNSSafe(n int) (string, error) {
-	str, err := GenerateRandomStringURLSafe(n)
-
-	str = strings.ToLower(strings.ReplaceAll(strings.ReplaceAll(str, "_", ""), "-", ""))
+	var str string
+	var err error
+	for {
```

Comment:
This infinite for loop with the break on the if statement on line 336 here can be changed to a while-style for loop instead:

```go
for len(str) < n {
    str, err = GenerateRandomStringURLSafe(n)
    if err != nil {
        return "", err
    }
    str = strings.ToLower(strings.ReplaceAll(strings.ReplaceAll(str, "_", ""), "-", ""))
}
```

Reply:
yep, updated